lib/route.js

```diff
@@ -211,22 +211,22 @@ function buildRouting (options) {
         method: opts.method
       }
 
-      const context = new Context(
-        opts.schema,
-        opts.handler.bind(this),
-        this[kReply],
-        this[kRequest],
-        this[kContentTypeParser],
-        config,
-        opts.errorHandler || this[kErrorHandler],
-        opts.bodyLimit,
-        opts.logLevel,
-        opts.logSerializers,
-        opts.attachValidation,
-        this[kReplySerializerDefault],
-        opts.schemaErrorFormatter || this[kSchemaErrorFormatter],
-        this
-      )
+      const context = new Context({
+        schema: opts.schema,
+        handler: opts.handler.bind(this),
+        Reply: this[kReply],
+        Request: this[kRequest],
+        contentTypeParser: this[kContentTypeParser],
+        config: config,
```

Comment:
all these parameters are no more necessary thanks to the `server` option.

Reply:
I think that is a separate issue. This one should focus on the refactor.

Reply:
@Eomm you meant something like this? [commit](https://github.com/fastify/fastify/pull/3483/commits/fe800b830311ff92702972302ddd19bbf6abb0c9)

We need to specify fields that are allowed to be used because context might pick unwanted fields.

Reply:
I do not think the function is actually needed.

```diff
diff --git a/lib/context.js b/lib/context.js
index bb36364..fd0a0b7 100644
--- a/lib/context.js
+++ b/lib/context.js
@@ -17,9 +17,6 @@ const {
 function Context ({
   schema,
   handler,
-  Reply,
-  Request,
-  contentTypeParser,
   config,
   errorHandler,
   bodyLimit,
@@ -28,23 +25,13 @@ function Context ({
   attachValidation,
   replySerializer,
   schemaErrorFormatter,
-  server,
:...skipping...
diff --git a/lib/context.js b/lib/context.js
index bb36364..fd0a0b7 100644
--- a/lib/context.js
+++ b/lib/context.js
@@ -17,9 +17,6 @@ const {
 function Context ({
   schema,
   handler,
-  Reply,
-  Request,
-  contentTypeParser,
   config,
   errorHandler,
   bodyLimit,
@@ -28,23 +25,13 @@ function Context ({
   attachValidation,
   replySerializer,
   schemaErrorFormatter,
-  server,
-  allowedServerFields
+  server
 }) {
-  const getAllowedValue = (passedValue, serverValue, fieldName) => {
-    const isServerFieldNotAllowed = !allowedServerFields.includes(fieldName)
-
-    if (isServerFieldNotAllowed) {
-      return passedValue
-    }
-    return typeof passedValue === 'undefined' ? serverValue : passedValue
-  }
-
   this.schema = schema
   this.handler = handler
-  this.Reply = getAllowedValue(Reply, server[kReply], 'Reply')
-  this.Request = getAllowedValue(Request, server[kRequest], 'Request')
-  this.contentTypeParser = getAllowedValue(contentTypeParser, server[kContentTypeParser], 'contentTypeParser')
+  this.Reply = server[kReply]
+  this.Request = server[kRequest]
+  this.contentTypeParser = server[kContentTypeParser]
   this.onRequest = null
   this.onSend = null
   this.onError = null
@@ -52,17 +39,17 @@ function Context ({
   this.preHandler = null
   this.onResponse = null
   this.config = config
-  this.errorHandler = getAllowedValue(errorHandler, server[kErrorHandler], 'errorHandler')
+  this.errorHandler = errorHandler || server[kErrorHandler]
   this._middie = null
   this._parserOptions = {
-    limit: getAllowedValue(bodyLimit, server[kBodyLimit], 'bodyLimit') || null
+    limit: bodyLimit || server[kBodyLimit]
   }
-  this.logLevel = getAllowedValue(logLevel, server[kLogLevel], 'logLevel')
+  this.logLevel = logLevel || server[kLogLevel]
   this.logSerializers = logSerializers
   this[kFourOhFourContext] = null
   this.attachValidation = attachValidation
-  this[kReplySerializerDefault] = getAllowedValue(replySerializer, server[kReplySerializerDefault], 'replySerializer')
-  this.schemaErrorFormatter = getAllowedValue(schemaErrorFormatter, server[kSchemaErrorFormatter], 'schemaErrorFormatter') || defaultSchemaErrorFormatter
+  this[kReplySerializerDefault] = replySerializer
+  this.schemaErrorFormatter = schemaErrorFormatter || server[kSchemaErrorFormatter] || defaultSchemaErrorFormatter
```

Reply:
thanks!